src/torchcodec/samplers/_implem.py

```diff
@@ -179,9 +184,24 @@ def to_framebatch(frames: list[Frame]) -> FrameBatch:
             data=data, pts_seconds=pts_seconds, duration_seconds=duration_seconds
         )
 
-    all_decoded_frames: list[Frame] = [
-        decoder.get_frame_at(index) for index in all_clips_indices
-    ]
+    all_clips_indices_sorted, argsort = zip(
+        *sorted((j, i) for (i, j) in enumerate(all_clips_indices))
```

Comment:
Nit: i, j makes both look like indexes in the same range. Maybe call them batch_index, frame_index?